src/bin/vip-sync.js

```diff
@@ -56,8 +56,22 @@ command( {
 					}
 				} );
 		} catch ( e ) {
-			syncing = true;
+			if ( e.graphQLErrors ) {
+				let bail = false;
+
+				for ( const err of e.graphQLErrors ) {
+					if ( err.message !== 'Site is already syncing' ) {
```

Comment:
Don't we have a way to play it safer by checking for a return value i.e. status code or anything more "fixed" rather than for a string (error message)?

Reply:
Unfortunately not really. I briefly addressed this point in the commit message for https://github.com/Automattic/vip/pull/144/commits/765574612530e85a7e343083017734b51d940263.

We want to bail out if there are any sync errors. The problem here is that we consider it to be an error in the API if you try to sync a site that's already syncing, but it's not an error in the CLI. It's related to the expectations being slightly different. In the API, we only care about the single request -- you tried to sync a site that already had an active sync. In the CLI, we can be somewhat more graceful and show you the status for the in-progress sync.

I don't think we want to change the response code (to 200, for example) in the API because the request is to start a sync and we're not doing that.

I think the best alternate solution would be to introduce some kind of error code to the API, but that's going to involve changing the response format (or introducing a different response format for this endpoint).